### PR TITLE
Kernel_23: Add an equality test to Aff_transfo 2 and 3.

### LINKED ISSUE #3055: Add an equality test to Aff_transformation_2 and Aff_transformation_3 classes
## Issue Details

Could we have an equality test for the affine transform classes Aff_transformation_2 and Aff_transformation_3?  This is very useful (perhaps essential) for writing tests for client code that creates and manipulates affine transforms.

## Environment

* Operating system (Windows/Mac/Linux, 32/64 bits): all
* Compiler: all
* Release or debug mode: all
* Specific flags used (if any):
* CGAL version: 4.1x
* Boost version: N/A
* Other libraries versions if used (Eigen, TBB, etc.): N/A

### PR DESCRIPTION
**Pull request by maxGimeno**
## Summary of Changes
Add an equality test to Aff_transformation_2 and Aff_transformation_3.

## Release Management

* Affected package(s): Kernle_23, Cartesian_kernel, Homogeneous_kernel
* Issue(s) solved (if any): fix #3055 
* Feature/Small Feature (if any):
* Link to compiled documentation (obligatory for small feature) [link](https://mydoc/Manual/Pkg)
* License and copyright ownership:

### CODE REVIEW

File: Kernel_23/test/Kernel_23/include/CGAL/_test_cls_aff_transformation_2.h [hunk 1]
```diff
@@ -577,6 +577,13 @@ _test_cls_aff_transformation_2(const R& )
  assert( ident.homogeneous(1,2) == ident.hm(1,2) );
  assert( gscale.homogeneous(1,1) == gscale.hm(1,1) );
 
+ //equality
+ CGAL::Aff_transformation_2<R> a2(0,1,0,1),
+     a3(0,1,0,1), a4(0,0,1,1);
+ assert(a2 == a3);
+ assert(a3 != a4);
```
Comment (lrineau) [inline Kernel_23/test/Kernel_23/include/CGAL/_test_cls_aff_transformation_2.h]: You should test `Aff_transformation_3` as well.

Comment (afabri) [reply Kernel_23/test/Kernel_23/include/CGAL/_test_cls_aff_transformation_2.h]: Don't we have to compare quotients for Homogeneous?

Comment (maxGimeno) [reply Kernel_23/test/Kernel_23/include/CGAL/_test_cls_aff_transformation_2.h]: @afabri the operator== compares the results of homogeneous(i,j), so it is taken into account.

**Comment by sloriot** [comment]
Successfully tested in [4.14-Ic-38](https://cgal.geometryfactory.com/CGAL/testsuite/results-4.14-Ic-38.shtml).

**Comment by sloriot** [comment]
However, the testsuite seems a bit weak. In particular, it seems that exact constructions are required. @lrineau Am I missing something?

**Comment by lrineau** [comment]
[**@sloriot**](https://github.com/sloriot) commented on [Nov 8, 2018, 5:17 PM GMT+1](https://github.com/CGAL/cgal/pull/3399#issuecomment-437055546 "2018-11-08T16:17:32Z - Replied by Github Reply Comments"):
> However, the testsuite seems a bit weak. In particular, it seems that exact constructions are required. @lrineau Am I missing something?

We are testing equality of constructed objects, here. Either we use trivial coordinates (full of zeros and ones), and we know that the construction in with `using FT = double;` will be exact, or we require a kernel with exact constructions. Maybe *I* missed something. How could it be tested differently?

**Comment by sloriot** [comment]
From what I understand there is a quotient that is explicitly constructed in `operator==()`.

**Comment by lrineau** [comment]
I do not know. Where is the construction of the quotient?

**Comment by sloriot** [comment]
My bad,  I thought `cartesian(int i, int j)` was computing a ratio but after looking at the implementation, it returns an internal coefficient. I'll merge the branch then.

**Pull request merged**